source/marsGameHud.vwf.yaml

```diff
@@ -0,0 +1,468 @@
+---
+extends: http://vwf.example.com/hud/overlay.vwf
+implements:
+  - http://vwf.example.com/sceneGetter.vwf
+properties:
+  visible: true
+  lastBlocklyNodeID:
+methods:
+  elementPreDraw:
+    parameters:
+      - context
+      - element
+    body: |
+      var alpha = 1;
+      var time = Date.now() / 1000;
+      if ( !element.enabled ) {
+        alpha = 0.5;
+      } else if ( element.isBlinking ) {
+        if ( time  - element.lastBlinkTime > element.blinkInterval ) {
+          alpha = 0.5;
+          if ( time - element.lastBlinkTime > element.blinkInterval + element.blinkDuration ) {
+            element.lastBlinkTime = time;
+          }
+        }
+      }
+      context.globalAlpha = alpha;
```

Comment:
Sort of. It helps when writing the draw functions to be able to see all of the properties available to you. The draw functions, by the way, are used only by the HUD driver. They're not true methods. I would like to find a way to make that clear, but I have not yet found a way. `this` in the draw function refers to the view object (HUD element) rather than the VWF node itself.